lineage/states/StateDistribution.py

```diff
@@ -38,15 +36,16 @@ def pdf(self, tuple_of_obs):  # user has to define how to calculate the likeliho
         # distribution observations), so the likelihood of observing the multivariate observation is just the product of
         # the individual observation likelihoods.
 
-        try:
-            bern_ll = bern_pdf(tuple_of_obs[0], self.bern_p)
-        except ZeroDivisionError:
-            assert False, f"{tuple_of_obs[0]}, {self.bern_p}"
+        bern_ll = bern_pdf(tuple_of_obs[0], self.bern_p) if tuple_of_obs[2] == 1 else 1.0
 
         try:
-            gamma_ll = gamma_pdf(tuple_of_obs[1], self.gamma_a, self.gamma_scale)
+            if tuple_of_obs[2] == 1:
+                gamma_ll = gamma_pdf(tuple_of_obs[1], self.gamma_a, self.gamma_scale)
+            else:
+                gamma_ll = sp.gamma.sf(tuple_of_obs[1], a=self.gamma_a, scale=self.gamma_scale)
```

Comment:
I agree with/like this approach. Knowing which observations are censored should be used when calculating the likelihood of that observation.